src/parse.rs

```diff
@@ -1266,8 +1266,8 @@ pub(crate) struct HtmlScanGuard {
     pub declaration: usize,
 }
 
-pub type BrokenLinkCallback<'a> =
-    Option<&'a mut dyn FnMut(BrokenLink) -> Option<(CowStr<'a>, CowStr<'a>)>>;
+pub type BrokenLinkCallback<'input, 'borrow> =
```

Comment:
This type signature looks good. How does this compare to the previous type? The old type can be expressed as `BrokenLinkCallback<'a, 'a>`, right? If so, this would be a strict supertype. (and therefore not breaking unless explicit lifetimes were used by users)

Reply:
Yes, the previous type was equivalent to `BrokenLinkCallback<'input, 'input>` (where `'input` is the lifetime of the markdown).

Reply:
> (and therefore not breaking unless explicit lifetimes were used by users)

Probably will be breaking for people who write top-level functions that take `BrokenLink<'_>` (*ahem*, me) unfortunately. But, it will be very easy to adapt to (just do `BrokenLink<'_, '_>`).

Reply:
@camelid `BrokenLink` isn't changing, just `BrokenLinkCallback`. This is a private type alias.

Reply:
Oops, somehow missed that :)

Reply:
It was made public to resolve https://github.com/raphlinus/pulldown-cmark/issues/498.